### PR TITLE
Alerting: Add metrics for remote secondary mode to the forked AM

### PR DESCRIPTION
**Pull request by santihernandezc**
This PR adds metrics around configuration and state sync attempts (total, failed).

### CODE REVIEW

File: pkg/services/ngalert/metrics/remote_secondary_forked_alertmanager.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	ConfigurationSyncsFailedTotal *prometheus.CounterVec
+	StateSyncsFailedTotal         *prometheus.CounterVec
+}
+
+func NewRemoteSecondaryForkedAlertmanagerMetrics(r prometheus.Registerer, subsystem string) *RemoteSecondaryForkedAlertmanager {
+	return &RemoteSecondaryForkedAlertmanager{
+		SyncsTotal: promauto.With(r).NewCounterVec(prometheus.CounterOpts{
+			Namespace: Namespace,
+			Subsystem: subsystem,
+			Name:      "remote_secondary_forked_alertmanager_syncs_total",
```
Comment (alexweav) [inline pkg/services/ngalert/metrics/remote_secondary_forked_alertmanager.go:19]: Do these metrics apply specific to the remote **secondary** alertmanager? They seem like they'd make sense for any remote alertmanager, even in primary mode.

If so, could we consider dropping `secondary` from all the names?

Also, `forked` feels like an implementation detail. Could it be `remote_alertmanager_syncs_total` and similar?

Comment (santihernandezc) [reply pkg/services/ngalert/metrics/remote_secondary_forked_alertmanager.go:19]: This is only for remote secondary, we won't sync config and state in other modes, but rather redirect writes to the remote Alertmanager. There won't be any time-based convergence.

Comment (gotjosh) [reply pkg/services/ngalert/metrics/remote_secondary_forked_alertmanager.go:19]: Agree with alex and I would take it one step further and and get rid of any sort of attachment to any of the modes. Metrics just exists as-is and are used by whatever mode we're on. 

As such, everything should be in the remote alertmanager metrics struct.

**Comment by santihernandezc** [comment]
Closing in favor of https://github.com/grafana/grafana/pull/79835

**Pull request closed without merging**